distributed/client.py

```diff
@@ -1596,28 +1602,13 @@ def close(self, timeout=no_default):
             return
         self.status = "closing"
 
-        with suppress(AttributeError):
-            for pc in self._periodic_callbacks.values():
-                pc.stop()
-
         if self.asynchronous:
             coro = self._close()
             if timeout:
                 coro = asyncio.wait_for(coro, timeout)
             return coro
 
-        if self._start_arg is None:
```

Comment:
A Client(asynchronous=False) closes its cluster, but an asynchronous one does not. This makes them more consistent and handles `@gen.coroutine`

Now that the `asynchronous=self._asynchronous` is consistent it should always be safe to `await self.cluster.close()` in the `Client._close`